dns/dkim/main.tf

```diff
@@ -0,0 +1,53 @@
+
+variable "sudoer_username" {}
+
+variable "public_instances" {}
+
+variable "ssh_private_key" {
+  type = string
+}
+
+resource "tls_private_key" "dkim" {
+  algorithm = "RSA"
+}
+
+resource "null_resource" "deploy_certs" {
+  for_each = var.public_instances
```

Comment:
Instead of deploying to all public instances, it would be simpler to deploy the certificate in `/etc/puppetlabs/code/environment/production/site/profiles/files/mail/` and have Puppet deploy the certificates where it is needed.